handlers/fresh.ts

```diff
@@ -12,8 +12,13 @@ export const isFreshCtx = <TState>(
   return typeof (ctx as HandlerContext).render === "function";
 };
 export default function Fresh(page: FreshConfig) {
-  return (_: Request, ctx: ConnInfo) =>
-    isFreshCtx(ctx)
+  return (req: Request, ctx: ConnInfo) => {
+    const url = new URL(req.url);
+    if (url.searchParams.get("asJson") !== null) {
+      return Response.json(page);
+    }
```

Comment:
if `?asJson` is provided, so instead of rendering the component a JSON object will be returned